xdp-dump/xdpdump.c

```diff
@@ -1603,14 +1603,23 @@ static bool capture_on_interface(struct dumpopt *cfg)
 	struct capture_programs      tgt_progs = {};
 
 	mp = xdp_multiprog__get_from_ifindex(cfg->iface.ifindex);
-	if (IS_ERR_OR_NULL(mp)) {
+	if (IS_ERR_OR_NULL(mp) ||
+	    (!IS_ERR_OR_NULL(mp) && xdp_multiprog__main_prog(mp) == NULL)) {
```

Comment:
Boolean logic short-circuiting makes that second `!IS_ERR_OR_NULL()` redundant...

Reply:
You are right, I normally rely on the compiler to do this. But in this case, it's obvious, will change.